src/app/binding/BindingTable.h

```diff
@@ -0,0 +1,104 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "app/OperationalDeviceProxy.h"
+#include "lib/core/DataModelTypes.h"
+#include "lib/dnssd/Resolver.h"
+#include "lib/support/Pool.h"
+
+namespace chip {
+namespace app {
+
+struct BindingTableEntry
+{
+    BindingTableEntry(EndpointId sourceEndpoint, NodeId peerNodeId, EndpointId peerEndpointId, FabricIndex peerFabricIndex,
+                      ClusterId clusterId);
+
+    EndpointId mSourceEndpoint;
+    EndpointId mPeerEndpointId;
+    PeerId mPeerId;
+    GroupId mPeerGroupId; // Destination group, not used
+    ClusterId mClusterId;
+};
+
+class BindingTable
+{
+public:
+    BindingTable();
+
+    CHIP_ERROR Add(EndpointId sourceEndpoint, NodeId peerNodeId, EndpointId peerEndpointId, FabricIndex peerFabricIndex,
+                   ClusterId clusterId);
+
+    CHIP_ERROR Remove(EndpointId sourceEndpoint, NodeId peerNodeId, EndpointId peerEndpointId, FabricIndex peerFabricIndex,
+                      ClusterId clusterId);
+
+    BindingTableEntry * FindBindingEntry(PeerId peer);
+
+    BindingTableEntry * FindBindingEntry(NodeId node, FabricIndex fabricIndex);
+
+    BindingTableEntry * FindBindingEntry(EndpointId endpoint, ClusterId cluster);
+
+    OperationalDeviceProxy * FindPeerDevice(PeerId peer);
+
+    OperationalDeviceProxy * FindPeerDevice(NodeId node, FabricIndex fabricIndex);
```

Comment:
These all need documenting.  Especially the version of FindBindingEntry that takes endpoint+cluster, and is not documented that this is the _local_ endpoint...